Spring colour one

AkashSaggar opened the pull request:
# Description 
Created 4 colour theme CSS files


# Type of change 
- [ ] Bug fix
- [x] New feature
- [ ] Refactoring
- [ ] Documentation 


# Problem
Briefly describe the root cause and analysis of the problem 


# Solution description
Describe your code changes in detail for reviewers. 
Explain the technical solution you have provided and how it fixes the issue.


# Tests 
All unit tests pass?

Builds successfully?


Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)


Provide steps in detail for reviewers to reproduce the expected result of the code changes.


# Relevant document/ link (if any) 
Any related documents could help reviewers to review this PR 


# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.

jfan0010 commented on .gitignore:
```diff
@@ -119,3 +119,6 @@ dist
 !.elasticbeanstalk/*.global.yml
+
+# Final static build files
+scholly/base/build
```
out of curiosity, why are we ignoring this file?

aksh0001 replied:
I believe this is the generated `build` after `npm run build` is executed. I think we don't want this in the git but I thought we ignored this already. Thanks for adding it

jfan0010 reviewed (commented):
might not be necessary, but I think it would be great if you can attach some screenshots of the themes you've created :)

aksh0001 commented on scholly/base/src/components/App.js:
```diff
@@ -6,6 +6,11 @@ import { Switch, Route } from 'react-router-dom';
 import './centred.css';
+import './baseColours.css';
+import './darkColours.css';
+//import './lightColours.css';
+//import './thirdTheme.css';
+//import './fourthTheme.css';
```
Please remove commented-out lines. We are planning to upgrade the SA for scholly base as well and this will throw a check.

aksh0001 replied:
If you'd like to document that these are the available css stylesheets, I think it's better done over a comment or in the file docs at the top.

AkashSaggar replied:
@aksh0001 just resolving this because we've decided on just 2 colours now so this should be all good. Dynamic switching will be implemented after PR 226 is merged.

AkashSaggar commented:
Major changes made recently:
- Removed colours 3 & 4 (just light and dark now)
- Updated so they work properly with the various layout options recently implemented
- After PR 226 is merged I can make it dynamic (just need to replace the const darkMode = true variable with an actual variable from data or fakeData)

jfan0010 commented on scholly/base/src/layouts/layout1/components/layout/Header.js:
```diff
@@ -16,8 +16,7 @@ const Header = () => {
         expand="md"
-        bg="light"
-        variant="light"
+        variant=""
```
Might be dumb: why do we need `variant=""` when there's no value in it?

AkashSaggar replied:
@jfan0010 Some of the components automatically assign a default variant (light or primary) if I don't have this (and that overwrites the colours I've assigned in CSS).
Navbar does this but some others don't for some reason (in those cases it's fine to delete the variant entirely).

jfan0010 commented on scholly/base/src/layouts/layout4/App.js:
```diff
@@ -6,8 +6,17 @@ import { Switch, Route } from 'react-router-dom';
+  import('../../shared/css/darkColours.css');
+} else {
+  import('../../shared/css/lightColours.css');
+}
```
Just a personal suggestion: since PR#223 has been approved and merging soon (I guess), I think it will be a good idea not to make any change in layout4, so you won't need to fix the conflict when that PR is deleting layout 4 entirely.

AkashSaggar replied:
@jfan0010 Yup makes sense. I've reverted my changes to layout 4 files.

rsal0007 reviewed (approved):
Looks good!

jfan0010 commented on scholly/base/src/layouts/layout1/App.js:
```diff
@@ -8,6 +8,15 @@ import { TEAM_INFO } from '../../global/data';
+import '../../shared/css/style.css';
+import '../../shared/css/baseColours.css';
+
+const darkMode = false;
```
Can you move the `darkMode = false` or something like `themeOption` into `fakeData.js` and import it here like how we import the layoutOption? I think that would be easier for others who are currently working on scholly frontend.

AkashSaggar replied:
@jfan0010 Kunj has added a FAKE_TEAM_SITE_METADATA in PR 226.
Didn't want to conflict with that, so I think it's better for me to use that after PR 226 is merged.

AkashSaggar replied:
@jfan0010 I've changed the boolean into a string for now anyway, so it will be more consistent with what Kunj is implementing. Although still haven't used fakeData.js for now to not conflict with PR 226 changes.

jfan0010 reviewed (approved):
Functionality-wise looks good to me! Not 100% sure if you guys have decided to use boolean or number options for selecting the theme. If we are going for the number options (or strings), might be a good idea to change the way we store theme in this PR.

vgoh1 commented on scholly/base/src/layouts/layout2/App.js:
```diff
@@ -6,10 +6,22 @@ import { Switch, Route } from 'react-router-dom';
+} else if (themeOption === '2') {
+  import('../../shared/css/darkColours.css');
+} else {
+  // Fallback to light mode if unkown theme option is used
```
spelling mistake here

AkashSaggar replied:
Thanks, this has been fixed

vgoh1 commented on scholly/base/src/layouts/layout3/App.js:
```diff
@@ -6,8 +6,20 @@ import { Switch, Route } from 'react-router-dom';
+} else if (themeOption === '2') {
+  import('../../shared/css/darkColours.css');
+} else {
+  // Fallback to light mode if unkown theme option is used
```
spelling mistake here

AkashSaggar replied:
Thanks, this has been fixed

vgoh1 commented:
Oops, hit approve a bit prematurely. Before you merge, could you add the theme option into fakeData.js so we can control it from there rather than in each of the App.js?
